### PR TITLE
Let `board listall` return boards sorted with the same order as in the original `boards.txt`

### PR DESCRIPTION
**Pull request by cmaglie**
## Please check if the PR fulfills these requirements

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

- [X] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [X] The PR follows [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

## What kind of change does this PR introduce?

The ordering of the boards returned by `board listall` is now the same as the order of definition in the `boards.txt` of the platform.

## What is the current behavior?

The ordering is undefined (random):

```
$ arduino-cli board listall arduino:avr --format json | jq '.boards[] | .name'
"Arduino Industrial 101"
"Arduino Esplora"
"Arduino Robot Motor"
"Arduino Nano"
"Arduino Yún"
"Arduino Mini"
"Arduino Uno WiFi"
"Arduino Duemilanove or Diecimila"
"Arduino Uno Mini"
"Adafruit Circuit Playground"
"Arduino Gemma"
"LilyPad Arduino"
"Arduino Mega ADK"
"Arduino Leonardo ETH"
"Arduino Micro"
"Arduino Robot Control"
"Arduino Pro or Pro Mini"
"Linino One"
"Arduino Ethernet"
"Arduino Mega or Mega 2560"
"Arduino Fio"
"Arduino BT"
"Arduino Leonardo"
"Arduino NG or older"
"Arduino Yún Mini"
"Arduino Uno"
"LilyPad Arduino USB"
```

## What is the new behavior?

The ordering is the same as in the `boards.txt`
```
$ arduino-cli board listall arduino:avr --format json | jq '.boards[] | .name'
"Arduino Yún"
"Arduino Uno"
"Arduino Uno Mini"
"Arduino Duemilanove or Diecimila"
"Arduino Nano"
"Arduino Mega or Mega 2560"
"Arduino Mega ADK"
"Arduino Leonardo"
"Arduino Leonardo ETH"
"Arduino Micro"
"Arduino Esplora"
"Arduino Mini"
"Arduino Ethernet"
"Arduino Fio"
"Arduino BT"
"LilyPad Arduino USB"
"LilyPad Arduino"
"Arduino Pro or Pro Mini"
"Arduino NG or older"
"Arduino Robot Control"
"Arduino Robot Motor"
"Arduino Gemma"
"Adafruit Circuit Playground"
"Arduino Yún Mini"
"Arduino Industrial 101"
"Linino One"
"Arduino Uno WiFi"
```

## Does this PR introduce a breaking change, and is [titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?

No

### CODE REVIEW

**Review by umbynos — APPROVED** [review]
Code LGTM, let's wait for a review by @AlbyIanna

**Comment by AlbyIanna** [comment]
I tested the command as described in the PR description and it works fine.
I have one more request though: would it be possible to get the boards listed in the same way when using the [`board details`](https://arduino.github.io/arduino-cli/0.26/commands/arduino-cli_board_details/) command? I'm asking it because [the Arduino IDE is using the `board details` command](https://github.com/arduino/arduino-ide/blob/main/arduino-ide-extension/src/node/boards-service-impl.ts#L68) (instead of `listall`) to get the list. This allows us to call the CLI only once to [get](https://github.com/arduino/arduino-ide/blob/main/arduino-ide-extension/src/node/boards-service-impl.ts#L96) [other](https://github.com/arduino/arduino-ide/blob/main/arduino-ide-extension/src/node/boards-service-impl.ts#L98) [valuable](https://github.com/arduino/arduino-ide/blob/main/arduino-ide-extension/src/node/boards-service-impl.ts#L107) [data](https://github.com/arduino/arduino-ide/blob/main/arduino-ide-extension/src/node/boards-service-impl.ts#L149-L150).

@cmaglie does it make sense to you? Otherwise we could just call the arduino-cli twice, once to get the ordered list and another time to get the other data.

**Comment by cmaglie** [comment]
Actually, the `BoardDetails` request returns details about **one** board, because you must specify the FQBN of that board:

```
message BoardDetailsRequest {
  // Arduino Core Service instance from the `Init` response.
  Instance instance = 1;
  // The fully qualified board name of the board you want information about
  // (e.g., `arduino:avr:uno`).
  string fqbn = 2;
}
```

this means that the IDE obtains the FQBN in some other ways (for example with a `BoardListAll` request or similar).
BTW I can extend the ordering to other calls too, but not the `BoardDetails` since it returns the details of just one board and there is nothing to sort there :-).

**Comment by AlbyIanna** [comment]
Sorry @cmaglie, my bad. You are right, we use the BoardDetails request for doing another thing.

We are actually using the [BoardSearch request](https://github.com/arduino/arduino-ide/blob/main/arduino-ide-extension/src/node/boards-service-impl.ts#L199) to get those packages.

I can easily make a change to use listall instead.

Thank you @cmaglie !

**Review by AlbyIanna — APPROVED** [review]
I gave the build a try with the IDE 2 and this was the result:
Screenshot 2022-10-04 at 10 51 36

Which, to me, looks pretty similar to the IDE 1 listing:
Screenshot 2022-10-04 at 10 51 46

LGTM 🚀 
Thanks @cmaglie !

**Pull request merged**
